src/gt4py/next/iterator/embedded.py

```diff
@@ -523,38 +523,33 @@ def promote_scalars(val: CompositeOfScalarOrField):
         )
 
 
+_python_builtins: dict[str, Callable] = {
```

Comment:
this change is unrelated (note I removed float and int from the list) and consider the TODO resolved. Will role back if this breaks something...